apps/web/src/stores/currentWorkspace.ts

```diff
@@ -42,8 +44,39 @@ export default function useCurrentWorkspace(opts?: SWRConfiguration) {
       mutate(workspace)
       return workspace
     },
-    [mutate],
+    [mutate, data],
+  )
+
+  const { execute: setDefaultProvider } = useServerAction(
+    setDefaultProviderAction,
+  )
+  const updateDefaultProvider = useCallback(
+    async (payload: { defaultProviderId: number | null }) => {
+      if (!data) return
+
+      const [workspace, error] = await setDefaultProvider({
+        workspaceId: data!.id,
+        defaultProviderId: payload.defaultProviderId,
+      })
+      if (error) {
+        toast({
+          title: 'Failed to update workspace default provider',
+          description: error.message,
+          variant: 'destructive',
+        })
+
+        return
+      }
+
+      toast({
+        title: 'Default provider updated',
+      })
+
+      mutate(workspace)
+      return workspace
+    },
+    [mutate, data],
```

Comment:
for next times you can use `useLatitudeAction` which encapsulates the onsuccess and onerror use cases, besides providing some nice defaults in terms of error reporting